src/services/entity_repository.js

```diff
@@ -177,17 +177,18 @@ export default class EntityRepository {
   }
 
   async storeBundle(bundle, storagePeriods) {
-    await this.db.collection('bundles').insertOne({metadata: {storagePeriods}, ...bundle});
+    await this.db.collection('bundles').insertOne({...bundle});
+    await this.db.collection('bundle_metadata').insertOne({bundleId: bundle.bundleId, storagePeriods});
```

Comment:
I'm not sure if this is an issue, but there is a chance that first call is fine and second fails, so we will have bundle without metadata. There are transactions in mongo v4, but I don't think i's worth changing version

Reply:
We have deployed mongo v4, the problem is that transactions are only available when using a replica set. A possible solution would be to capture the id of the first insert and delete the record if the second insert fails.

Reply:
I would instead make the method in a way as to handle a second try nicely. That is with the current implementation if the first attempt fail on the second call, then a second attempt will fail on the first because of the unique requirement. So a `findOne and insertOne if not present` pattern would be advised.